src/magic_spell_effect.cpp

```diff
@@ -804,6 +806,24 @@ void spell_effect::spawn_summoned_monster( const spell &sp, Creature &caster,
     }
 }
 
+void spell_effect::spawn_summoned_vehicle( const spell &sp, Creature &caster,
+        const tripoint &target )
+{
+    if( g->m.veh_at( target ) ) {
+        add_msg( m_bad, _( "There is already a vehicle there." ) );
+        return;
+    }
+    vehicle *veh = g->m.add_vehicle( vproto_id( sp.summon_vehicle_id() ), target, -90, 100, 0 );
+    if( veh != nullptr ) {
+        veh->magic = true;
+        const time_duration summon_time = sp.duration_turns() * 100;
```

Comment:
don't forget the PERMANENT spell flag! make sure this spell checks for that when it's cast. I don't have the time to check it right this minute, but what happens if you try to spawn a multitile vehicle adjacent to a single tile vehicle? there's not a vehicle at that target, but you still would try to add_vehicle.

Reply:
> but what happens if you try to spawn a multitile vehicle adjacent to a single tile vehicle

`add_vehicle` returns a null pointer in that case.

Reply:
aha, so a more proper way of doing this is if add_vehicle == nullptr instead of the previous if statement